platform/android/export/export_plugin.cpp

```diff
@@ -525,6 +530,22 @@ bool EditorExportPlatformAndroid::is_package_name_valid(const String &p_package,
 		return false;
 	}
 
+	if (p_package.find("$genname") >= 0 && !is_project_name_valid(p_package)) {
+		if (r_error) {
+			*r_error = TTR("The end result may be unexpected. Try to explicitly specify the project name instead of the '$genname'.");
```

Comment:
Can you update the error message to:
`The project name does not meet the requirement for the package name format. Please explicitly specify the package name.`

Reply:
Done.